flowauth/frontend/src/TokenDetails.jsx

```diff
@@ -109,18 +110,18 @@ class TokenDetails extends React.Component {
     this.setState({ name: event.target.value });
   };
 
-  async componentDidMount() {
-    const rights = getMyRightsForServer(this.props.serverID);
-    try {
+  componentDidMount() {
+    getMyRolesOnServer(this.props.serverID)
+    .then((roles) => {
+      const scopes = scopes_with_roles(roles);
+      const expiry = scopes.map(d => d.latest_token_expiry).reduce((p,c) => p > c ? p : c)
       this.setState({
-        rights: (await rights).allowed_claims,
-        permitted: (await rights).allowed_claims,
-        expiry: (await rights).latest_expiry,
-        latest_expiry: (await rights).latest_expiry,
+        rights: scopes,
```

Comment:
I'd be tempted to rename the property too so it's the same as the value.

Reply:
Two different properties, `rights` and `permitted`, that start from the same root `scopes`